arkouda/array_api/linalg.py

```diff
@@ -1,36 +1,15 @@
 from .array_object import Array
 
-from arkouda.client import generic_msg
-from arkouda.pdarrayclass import create_pdarray, broadcast_if_needed
-
 
 def matmul(x1: Array, x2: Array, /) -> Array:
     """
     Matrix product of two arrays.
     """
-    from .array_object import Array
-
-    if x1._array.ndim < 2 and x2._array.ndim < 2:
-        raise ValueError(
-            "matmul requires at least one array argument to have more than two dimensions"
-        )
-
-    x1b, x2b, tmp_x1, tmp_x2 = broadcast_if_needed(x1._array, x2._array)
```

Comment:
I don't think ak_matmul does broadcasting. Could you add that there as a part of this PR?

Reply:
I made the change.  Let me know what you think.

Reply:
Looks good, thanks!